src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -195,66 +195,93 @@ public override void VisitParenthesizedLambdaExpression(ParenthesizedLambdaExpre
 
         public override void VisitLocalFunctionStatement(LocalFunctionStatementSyntax node)
         {
-            var body = (CSharpSyntaxNode)node.Body ?? node.ExpressionBody;
-            LocalFunctionSymbol match = null;
-            // Don't use LookupLocalFunction because it recurses up the tree, as it
-            // should be defined in the directly enclosing block (see note below)
+            LocalFunctionSymbol match = FindLocalFunction(node, _enclosing);
 
-            Binder possibleScopeBinder = _enclosing;
-            while (possibleScopeBinder != null && !possibleScopeBinder.IsLocalFunctionsScopeBinder)
-            {
-                possibleScopeBinder = possibleScopeBinder.Next;
-            }
+            bool oldSawYield = _sawYield;
+            _sawYield = false;
 
-            if (possibleScopeBinder != null)
+            BlockSyntax blockBody = node.Body;
+            ArrowExpressionClauseSyntax arrowBody = node.ExpressionBody;
+            var body = (CSharpSyntaxNode)blockBody ?? arrowBody;
+
+            InMethodBinder inMethodBinder = null;
+            if (body != null)
             {
-                foreach (var candidate in possibleScopeBinder.LocalFunctions)
+                VisitLocalFunctionBody(node, body, match, ref inMethodBinder);
+
+                if (_sawYield)
                 {
-                    if (candidate.Locations[0] == node.Identifier.GetLocation())
-                    {
-                        match = candidate;
-                    }
+                    _methodsWithYields.Add(body);
                 }
             }
 
-            bool oldSawYield = _sawYield;
-            _sawYield = false;
+            if (blockBody != null && arrowBody != null)
+            {
+                // There is an extraneous arrow body. We'll create binders for it too, re-using the inMethodBinder.
+                // We ignore whether this one has "yield".
+                VisitLocalFunctionBody(node, arrowBody, match, ref inMethodBinder);
```

Comment:
>VisitLocalFunctionBody(node, arrowBody, match, ref inMethodBinder); [](start = 16, length = 67)

It looks like we might be calling this method twice for the arrowBody when it is the only body in the syntax.